runtime/foucoco/src/xcm_config.rs

```diff
@@ -240,7 +194,28 @@ impl ShouldExecute for DenyReserveTransferToRelayChain {
 	}
 }
 
-pub type Barrier = AllowUnpaidExecutionFrom<Everything>;
+match_types! {
+	pub type ParentOrParentsPlurality: impl Contains<MultiLocation> = {
+		MultiLocation { parents: 1, interior: Here } |
+		MultiLocation { parents: 1, interior: X1(Plurality { .. }) }
+	};
+}
+
+pub type Barrier = (
```

Comment:
This change is not required for this PR. Did you just do it to harmonize Foucoco with the other two runtimes?

Reply:
Yes, that was the intention. Since Foucoco was accepting all messages freely. Of course we can decide to keep it that way. But I think it is good so we are able to test xcm configurations before they reach Amplitude.